pallets/auctions/src/lib.rs

```diff
@@ -0,0 +1,561 @@
+            match current_auction.bidders.find_bid(&who) {
+                // transfer funds to caller
+                Some(user_bid) => {
+                    T::CurrencyCaps::transfer(&Self::account_id(), &who, user_bid.1, KeepAlive)
+                }
+                None => Err(Error::<T>::BidDoesNotExist.into()),
+            };
+
+            // emit bid removed event
+            Self::deposit_event(Event::BidRemoved {
+                nft_id,
+                marketplace_id: current_auction.marketplace_id,
+                creator: who,
+            });
+
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        pub fn increase_bid(
+            origin: OriginFor<T>,
+            nft_id: NFTId,
+            amount: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+            let current_block = frame_system::Pallet::<T>::block_number();
+
+            // fetch data of auction that lists NFT
+            let current_auction =
+                Auctions::<T>::get(nft_id).ok_or(Error::<T>::AuctionDoesNotExist)?;
+
+            // ensure the auction period has not ended
+            ensure!(
+                current_auction.end_block > current_block,
+                Error::<T>::AuctionEnded
+            );
+
+            // ensure the bid is larger than the current highest bid
+            if let Some(current_highest_bid) = current_auction.bidders.get_current_highest_bid() {
+                ensure!(amount > current_highest_bid.1, Error::<T>::InvalidBidAmount);
+            }
+
+            // transfer funds from caller (subtracting amount from previous bid)
+            match current_auction.bidders.find_bid(&who) {
+                // transfer funds to caller
+                Some(user_bid) => {
+                    let amount_to_transfer = amount.checked_sub(&user_bid.1);
+                    T::CurrencyCaps::transfer(&who, &Self::account_id(), amount, KeepAlive)
+                }
+                None => Err(Error::<T>::BidDoesNotExist.into()),
+            };
+
+            // emit bid created event
+            Self::deposit_event(Event::BidUpdated {
+                nft_id,
+                marketplace_id: current_auction.marketplace_id,
+                creator: who,
+            });
+
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        pub fn buy_it_now(
+            origin: OriginFor<T>,
+            nft_id: NFTId,
+            amount: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+            let current_block = frame_system::Pallet::<T>::block_number();
+
+            // fetch the data of given nftId
+            let nft_data = T::NFTHandler::get_nft(nft_id).ok_or(Error::<T>::NFTIdInvalid)?;
+
+            // fetch data of auction that lists NFT
+            let current_auction =
```

Comment:
`current_auction` implies that we deal with multiple auctions at the same time hence the need for `current` prefix. I think  `auction` is sufficient.

Reply:
This comment is same for all the places where you use current_auction